### PR TITLE
Better README

### PR DESCRIPTION
**Pull request by vickysharma0812**
## Pull request purpose

related to issue #382, and enhancement in documentation.

## Detailed changes proposed in this pull request

- README file in markdown form
- Please check the headings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,185 +1,268 @@
...
 
-    find_package(arpack-ng)
-    add_executable(main main.f)
-    target_include_directories(main PUBLIC PARPACK::PARPACK)
-    target_link_libraries(main PARPACK::PARPACK)
+> The above script will builds everything including examples and parallel support (with MPI).
+
+### Using ARPACK-NG in Cmake project
+
+You can use arpack in your CMake builds by using `ARPACK::ARPACK` target. For example,
```
Comment (fghoussen) [inline README.md:173]: Should add the `CMAKE_MODULE_PATH` env variable must be updated by the user (or `find_package` won't find `arpack-ng` cmake file).

File: README.md [hunk 2]
```diff
@@ -1,185 +1,268 @@
...
-    
-    $ LIBSUFFIX="64" INTERFACE64="1" ITF64SUFFIX="ILP64" ./configure; make all install
-    $ cmake -DLIBSUFFIX="64" -DINTERFACE64=ON -DITF64SUFFIX="ILP64" ..; make all install
+
+```bash
+mkdir build
+cd build
+cmake -DCOVERALLS=ON -DCMAKE_BUILD_TYPE=Debug ..
+make all check test coveralls
+```
```
Comment (fghoussen) [inline README.md:206]: Coverage section can be killed: doesn't work anymore... @sylvestre: OK with that?

File: README.md [hunk 3]
```diff
@@ -1,185 +1,268 @@
...
+
+To get `ISO_C_BINDING` support:
```
Comment (fghoussen) [inline README.md:208]: For me, we want to turn this into a "ISO C BINDING" section at the same level that "ILP 64 support " sections and friends above.

File: README.md [hunk 4]
```diff
@@ -1,185 +1,268 @@
...
+Within DOCUMENTS directory there are three files for templates on how to invoke the computational modes of ARPACK.
+
+- ex-sym.doc
+- ex-nonsym.doc and
+- ex-complex.doc
+
+Also look in the README.MD file for explanations concerning the
+other documents.
+
+### Custom installation examples
```
Comment (fghoussen) [inline README.md:235]: Should be a subsection inside the "Install" section?

File: README.md [hunk 5]
```diff
@@ -1,185 +1,268 @@
...
+
+```bash
+    LIBSUFFIX="64" INTERFACE64="1" ITF64SUFFIX="ILP64" ./configure; make all install
+```
+
+```bash
+    cmake -DLIBSUFFIX="64" -DINTERFACE64=ON -DITF64SUFFIX="ILP64" ..; make all install
+```
+
+### Using with Intel-MKL
```
Comment (fghoussen) [inline README.md:261]: Move to "Install" section before "autotools" and "cmake" subsections?

File: README.md [hunk 6]
```diff
@@ -1,185 +1,268 @@
...
+other documents.
+
+### Custom installation examples
+
+```bash
+  LIBSUFFIX="64" ./configure; make all install
+```
+
+```bash
+  cmake -DLIBSUFFIX="64" ..; make all install
```
Comment (fghoussen) [inline README.md:242]: Remove leading spaces? Would look better?

File: README.md [hunk 7]
```diff
@@ -1,185 +1,268 @@
...
+```fortran
+  mkdir build
+  cd build
+  cmake -D EXAMPLES=ON -D MPI=ON -D BUILD_SHARED_LIBS=ON ..
+  make
+  make install
+```
 
-To use parpack from CMake, use PARPACK::PARPACK target:
+> Do not miss the `..` in line 3, as it denotes the location of source files.
```
[the remark this answers is not in the document]
Comment (vickysharma0812) [reply README.md:167]: I think we should keep it. It is very helpful in using the arpack lib. But, I have shifted it in `Extras`.

**Comment by fghoussen** [comment]
What is the purpose of `install.sh`?

**Comment by vickysharma0812** [comment]
> 

Please ignore it. It is for me to install ARPACK for my development.

**Comment by fghoussen** [comment]
Can you had commits for each remark above (easier to follow-up) before merging.

> Please ignore it. It is for me to install ARPACK for my development.

I can't, you added it in the PR: please, revert it or push-force.

**Comment by vickysharma0812** [comment]
@fghoussen  I have deleted the install.sh. Is there anything pending from myside?

**Comment by fghoussen** [comment]
Yes, can you had 1 commit for each remark (easier to follow-up) I made in the review before merging.

File: README.md [hunk 8]
```diff
@@ -1,185 +1,281 @@
...
 
-   Example programs for banded, complex, nonsymmetric, symmetric,
-   and singular value decomposition may be found in the directories
-   BAND, COMPLEX, NONSYM, SYM, SVD respectively.  Look at the README
-   file for further information.  To get started, get into the SIMPLE
-   directory to see example programs that illustrate the use of ARPACK in
-   the simplest modes of operation for the most commonly posed
-   standard eigenvalue problems.
 
+### Using auto-tools
```
Comment (sylvestre) [inline README.md:148]:
```suggestion
### Using autotools
```

**Comment by fghoussen** [comment]
@vickysharma0812: Yes, can you had 1 commit for each remark (easier to follow-up) I made in the review before merging.

**Comment by fghoussen** [comment]
We are about to create a new release #394: @vickysharma0812, do you want to finish this PR (https://github.com/opencollab/arpack-ng/pull/383#issuecomment-1363228510)? If not now, will be for next release.

**Comment by vickysharma0812** [comment]
> We are about to create a new release https://github.com/opencollab/arpack-ng/issues/394: @vickysharma0812, do you want to finish this PR (https://github.com/opencollab/arpack-ng/pull/383#issuecomment-1363228510)? If not now, will be for next release.

Honestly speaking, I am confused about what should I do. Can you please tell me, I am sorry for the inconvenience.

**Comment by fghoussen** [comment]
> Honestly speaking, I am confused about what should I do. Can you please tell me, I am sorry for the inconvenience.

You proposed a PR to make README more readable. Before merging, I asked you to apply several modifications (example: https://github.com/opencollab/arpack-ng/pull/383#discussion_r1045070390) you reacted to :+1: : now, you should add 1 commit for each remark above (the last one behing https://github.com/opencollab/arpack-ng/pull/383#discussion_r1045071799).

Also please rebase  your PR branch on master

**Comment by fghoussen** [comment]
@vickysharma0812: just realized you add modifications. Missed that, sorry.
All has been merged in #397

**Pull request closed without merging**